### PR TITLE
Fix fragment codegen.

### PR DESCRIPTION
**Pull request by mgilson**
## Description

Need to find the fields in the correct type (which is the type that the fragment is referencing.

Sadly the tests didn't pick this up because the fragment in the test is on a type called "Query".  Oops.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Matt Gilson for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/codegen/query_codegen.py [hunk 1]
```diff
@@ -198,14 +198,13 @@ def _populate_fragment_types(self, ast: DocumentNode) -> None:
             if isinstance(definition, FragmentDefinitionNode)
         )
         for fd in fragment_definitions:
-            query_type = self.schema.get_type_by_name("Query")
-            assert isinstance(query_type, TypeDefinition)
-
-            def graph_ql_object_type_factory(name: str, on: str):
-                return GraphQLFragmentType(name, on=on)
+            query_type = self.schema.get_type_by_name(fd.type_condition.name.value)
```
Comment (patrick91) [inline strawberry/codegen/query_codegen.py:201]: I should have noticed this, sorry!

**Review by patrick91 — APPROVED** [review]
Do you think we should add another test for this case?

**Comment by mgilson** [comment]
> Do you think we should add another test for this case?

It's easy enough to exercise this case with the test resources we have if we just change around what the query is returning.  I've updated it.

**Review by patrick91 — APPROVED** [review]
Thanks!

**Pull request merged**